ledger/ledger.go

```diff
@@ -795,6 +795,16 @@ func (l *Ledger) StartEvaluator(hdr bookkeeping.BlockHeader, paysetHint, maxTxnB
 		})
 }
 
+// FlushCaches flushes any pending data in caches so that it is fully available during future lookups.
+func (l *Ledger) FlushCaches() {
+	l.accts.accountsMu.Lock()
+
+	l.accts.baseAccounts.flushPendingWrites()
+	l.accts.baseResources.flushPendingWrites()
+
+	l.accts.accountsMu.Unlock()
```

Comment:
delegate to `l.accts`.

shouldn't we do the same for online accounts' base cache?

Reply:
Yes and no, the `writePending` isn't actually used in the `lruOnlineAccounts` struct. 
We are only doing protected `write`'s during `postCommit`.

Reply:
```suggestion
	l.accts.flushCaches()
```